pkg/engine/generation.go

```diff
@@ -14,17 +14,21 @@ import (
 //    - the caller has to check the ruleResponse to determine whether the path exist
 // 2. returns the list of rules that are applicable on this policy and resource, if 1 succeed
 func Generate(policyContext *PolicyContext) (resp *response.EngineResponse) {
-	return filterRules(policyContext)
+	policyStartTime := time.Now()
+	return filterRules(policyContext, policyStartTime)
 }
 
-func filterRules(policyContext *PolicyContext) *response.EngineResponse {
+func filterRules(policyContext *PolicyContext, startTime time.Time) *response.EngineResponse {
```

Comment:
Do we need to pass it in? Why don't we initialize startTime inside this function?

Reply:
Yep I agree with you but semantically, it seemed more apt that the startTime(the instant when the policy jussttt starts execute) is declared exactly as the first instruction of the policy's execution which is happening at 

```go
func Generate(policyContext *PolicyContext) (resp *response.EngineResponse) {
	policyStartTime := time.Now()
	return filterRules(policyContext, policyStartTime)
}
```